### PR TITLE
NumberBox: input should not reset the sign if minus is specific character (T920936)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

**Comment by alexander-kotov-dx** [comment]
it does not work for the accounting format example of [our accounting format demo](https://js.devexpress.com/Demos/WidgetsGallery/Demo/NumberBox/Formatting/jQuery/Light/)
Format in demo: "$ #,##0.##;($ #,##0.##)"

**Review by alexander-kotov-dx — CHANGES_REQUESTED** [review]
it does not work for the accounting format example of [our accounting format demo](https://js.devexpress.com/Demos/WidgetsGallery/Demo/NumberBox/Formatting/jQuery/Light/)
Format in demo: "$ #,##0.##;($ #,##0.##)"

**Review by San4es — COMMENTED** [review]
We have failed tests:
https://github.com/DevExpress/DevExtreme/pull/14153/checks?check_run_id=958331432

File: js/localization/number.js [hunk 1]
```diff
@@ -235,6 +235,21 @@ const numberLocalization = dependencyInjector({
         });
     },
 
+    getNegativeEtalonRegExp: function(format) {
```
Comment (DokaRus) [inline js/localization/number.js:238]: Could you add tests for [localization/ldml](https://github.com/DevExpress/DevExtreme/blob/20_2/testing/tests/DevExpress.localization/ldml.tests.js#L264)?

Comment (ksercs) [reply js/localization/number.js:238]: added

**Review by DokaRus — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**